File: mouse_clicker.c
#include <stdio.h>
#include <stdlib.h>
#include <windows.h>

// Function to simulate a mouse click at a given position
void autoClick(int x, int y) {
    // Infinite loop for continuous clicking
    while (1) {
        // Set the mouse position
        SetCursorPos(x, y);
        
        // Simulate mouse down and mouse up events (equivalent to a click)
        mouse_event(MOUSEEVENTF_LEFTDOWN, 0, 0, 0, 0);
        mouse_event(MOUSEEVENTF_LEFTUP, 0, 0, 0, 0);
        
        // Sleep for 100 milliseconds (0.1 seconds)
        Sleep(100);

        // Check if the ESC key is pressed
        if (GetAsyncKeyState(VK_ESCAPE)) {
            printf("Escape key pressed. Exiting.\n");
            break;
        }
    }
}

int main() {
    // Set the target position (example: center of a 4K screen)
    int x_position = 100; // x coordinate
    int y_position = 100; // y coordinate

    printf("Press the Escape key to stop.\n");
    
    // Call the autoClick function with the specified coordinates
    autoClick(x_position, y_position);

    return 0;
}

File: mouse_clicker.py
import pyautogui
import keyboard
import time

def auto_click(x, y):
    try:
        while True:
            # Move the mouse to the specified position
            pyautogui.moveTo(x, y)
            # Perform a mouse click
            pyautogui.click()
            # Wait for a short period to avoid overloading
            time.sleep(0.1)
            # Check if the Escape key has been pressed
            if keyboard.is_pressed('esc'):
                print("Escape key pressed. Exiting.")
                break
    except KeyboardInterrupt:
        print("Program interrupted by user. Exiting.")

if __name__ == "__main__":
    # Set the target position (example: center of a 4K screen)
    # Centrer :x_position = 1980,y_position = 1080
    #x_position = 2260  # x coordinate
    #y_position = 1800  # y coordinate
    #x_position = 1980  # x coordinate
    #y_position = 1950  # y coordinate
    x_position = 1980
    y_position = 1080
    #x_position = 1980
    #y_position = 1380
    #x_position = 1480
    #y_position = 1300
    #x_position = 1480
    #y_position = 1480

    print("Press the Escape key to stop.")
    auto_click(x_position, y_position)
